### ISSUE TITLE
Database examineHasLocalAlter failed convert current entity to JSON tree.

### ISSUE THREAD

**Issue by OnePoundd** [MassiveCraft/MassiveCore#310]
**Describe your environment**

* Affected Plugin version: 2.10
* MassiveCore version: 2.10
* Spigot version: 1.11
* Config pastebin: NOCHANGE

**Steps to reproduce this bug**

  1. Randomly occurs when the server has been running for a few minutes. Any faction activity could cause the issue or no activity could cause the issue, its pretty odd...

**Observed results**

Console is spammed with errors hundreds of times a second by massivecraft
https://hastebin.com/eqihihedig.md

**Expected results**

**Comment by OnePoundd** [comment]
Have since updated to factions 2.12.1, and moved over most of my new features. Some of my features relate heavily to the MPlayer entity class.

Whenever a new player joins, an MPlayer file is not created. Additionally at random intervals the console will begin to be spammed with the same has the hastebin above. I believe this happens when trying to export the updated MPlayer file.
When you create a faction, the MPlayer file is created, but as soon as the faction is disbanded and/or the player leaves the faction, their MPlayer file is deleted, and the error would start to happen at random intervals once again (if the player had a faction then the error would not get spammed).

I have only added fields to the MPlayer class of types such as long, int, Boolean and string, but I have recently added a "MassiveMapDef<String, PS>", to store custom locations related to the player and accessible by name. Anyway, this is what I believe to be the problem trigerring the "failed to convert current entity to JSON tree", but I can't put together why this would be the case when I use the exact same method to add 'faction warps' to the Faction Entity class.

Any help is greatly appreciated, I understand this may not fall under your supported topics as I'm editing your plugin and this would not be an issue for yourselves but I'm just curious as to why this would happen, and how I could possibly fix it.

TL;DR: When a player joins, no mplayer file is created, until they make/join a faction. Once they leave a faction or the faction is deleted the mplayer file is also deleted. Additionally, a 'failed to convert entity to JSON tree' issue is spammed in console when the player does not have a mplayer file. I believe this could be related to trying to add fields to the MPlayer entity (variable type "MassiveMapDef<String, PS>", however the same fields in the Faction entity class work and are exported absolutely fine.

**Comment by OnePoundd** [comment]
Done some more testing, if the MPlayer file is created before the faction is created, then it will remain once the faction is deleted... So there are no problems there at all. 

However for whatever reason a mplayer file is not created before the faction is created/deleted then the issues as described above will continue to occur.

When a new player joins the server, no mplayer file is created for them. And when you delete their mplayer file then a new file is not created for them. A mplayer file is only created when the faction is created, but then destroyed when the faction is left or disbanded.

I would like to know what events cause an mplayer file to be created, and for what reasons would this normal method not happen. It may well have something to do with the 'mplayer entity' not being able to  be converted into a 'JSON tree', but I would like to know for what reasons this would occur considering the only fields the mplayer has are Booleans, longs, strings, ints, and a "MassiveMapDef<String, PS>", which all work fine with the faction entities and their outputted files. You can see a more in-depth look of the error here:
https://hastebin.com/eqihihedig.md

As I said I understand this could be an issue with my code and possibly not covered by your support in general but I wouldn't be posting if I didn't have to, hopefully you guys can spot the issue in the hastebin/console log

**Comment by OnePoundd** [comment]
Just tested all this with the default factions plugin, no code changes/additions made. The same issue still occurs where no mplayer file is created on first login, or anything.

When you create a faction an mplayer file is created for that player but when you delete the faction it is also deleted. I'd like to know why this happens as its sort of dumb considering there are fields in the mplayer class that relate to the player, not the faction they are in. I would also like to know what I can do to ensure a mplayer file is created and that it remains after all sort of activities.
